### DISCUSSION TITLE
Custom error handling?

### DISCUSSION THREAD

**Discussion by ghost** [zino-app/graphql-flutter#1105] [Q&A] [answered]
Hello,

my backend sends customized formatted non standardized apollo exceptions, for example like this:

```json
{
  "statusCode": 400,
  "errors": [
    {
      "property": "email",
      "children": [],
      "constraints": {
        "isEmail": "Email must be an email"
      }
    },
    {
      "property": "password",
      "children": [],
      "constraints": {
        "matches": "Password must contain at least one number, one uppercase letter, one lowercase letter and one of the special characters (@ and _).",
        "minLength": "Password with value: \"123455\" needs to be at least 8 characters"
      }
    }
  ],
  "timestamp": "2022-04-17T15:32:14+02:00",
  "ip": "::1"
}
```

Is there way I can tell the client to use custom exception formatter?

**Comment by budde377** [comment, marked as answer]
Assuming that you're using `HttpLink`, you can use custom error parsing by implementing your own version of `ResponseParser` ([here](https://github.com/gql-dart/gql/blob/master/links/gql_link/lib/src/response_parser.dart#L4)) and pass this to the link.

**Reply by asainama** [reply]
Hey,Can you do an example, I can't figure it out? I have an HttpLinkServerException which throws before.
